ui/selectors/assets.ts

```diff
@@ -509,3 +511,43 @@ export const getMultichainNativeTokenBalance = createDeepEqualSelector(
     return balances[nativeAssetType];
   },
 );
+
+export const getAssetsBySelectedAccountGroup = createDeepEqualSelector(
+  ({ metamask }) => {
+    const initialState = {
+      accountTree: metamask.accountTree,
+      internalAccounts: metamask.internalAccounts,
+      allTokens: metamask.allTokens,
+      allIgnoredTokens: metamask.allIgnoredTokens,
+      tokenBalances: metamask.tokenBalances,
+      marketData: metamask.marketData,
+      currencyRates: metamask.currencyRates,
+      currentCurrency: metamask.currentCurrency,
+      networkConfigurationsByChainId: metamask.networkConfigurationsByChainId,
+      accountsByChainId: metamask.accountsByChainId,
+    };
+
+    let multichainState = {
+      accountsAssets: {},
+      assetsMetadata: {},
+      balances: {},
+      conversionRates: {},
+    };
+
+    ///: BEGIN:ONLY_INCLUDE_IF(keyring-snaps)
+    multichainState = {
+      accountsAssets: metamask.accountsAssets,
+      assetsMetadata: metamask.assetsMetadata,
+      balances: metamask.balances,
+      conversionRates: metamask.conversionRates,
+    };
+    ///: END:ONLY_INCLUDE_IF
```

Comment:
Code fence needed, as these controllers are initialised only behind that code fence.